### ISSUE TITLE
commentFormatting: it keeps asking to add spaces 

### ISSUE THREAD

**Issue by frederikhors** [go-critic/go-critic#1021]
It keeps asking to add spaces in this code:

```go
// var customTpl = `
//package {{.PkgName}}
//
```

`commentFormatting: put a space between `//` and comment text (gocritic)`

Why?

**Comment by quasilyte** [comment]
In Go, `//` comments without spaces are usually considered to be "directives".

Some examples:
```
//nolint
//go:noinline
//go:generate
```
We don't have a strict definition of a directive, but there is a convention, that comments that are directed to humans have a space between `//` and the comment text.

Note that in your exact case, it looks like it's not a comment, but rather a commented-out code. It should be reported by `commentedOutCode` diagnostic, but maybe it's disabled in your configuration.

**Comment by frederikhors** [comment]
Nothing is disabled: all is enabled in golangci-lint.

I'm using Goland and it doesn't put space betwen `//` and code it comments.

This is the problem.

But go-critic warns me even after I put the space in the first line (as you can see), why?

**Comment by quasilyte** [comment]
> But go-critic warns me even after I put the space in the first line (as you can see), why?

This is due to the fact that we use a comment node as a location for the warning.
If any of the comment block lines is triggering the warning, it will report an issue at the comment beginning.

> I'm using Goland and it doesn't put space betwen // and code it comments.

This is configurable. This issue tracker is not the best place to discuss that detail.

> Nothing is disabled: all is enabled in golangci-lint.

I'm not sure golangci-lint enables all go-critic checkers by default. In fact, it only enables ~40% of go-critic checkers.

> This is the problem.

Please re-read my previous answer. There is a reason why `commentFormatting` exists, I tried to explain it. If you don't like it, you can disable the diagnostic in your golangci lint configuration file. There is no bug here, it works as intended.
